platform/linux_platform.go

```diff
@@ -1062,6 +1063,25 @@ func (p linux) SetupLogDir() error {
 	return nil
 }
 
+func (p linux) SetupOptDir() error {
+	optDir := "/var/opt"
+
+	boshRootOptDirPath := path.Join(p.dirProvider.DataDir(), "root_opt")
+	err := p.fs.MkdirAll(boshRootOptDirPath, userRootOptDirPermissions)
```

Comment:
I have two questions here:

1. is it by intention to ignore the error? Other places react on this e.g. https://github.com/cloudfoundry/bosh-agent/blob/8d18e4294e235edeb7f7fefc2f081ce006f57a71/platform/linux_platform.go#L986-L988
2. if the `boshRootOptDirPath` already exists `MkdirAll` won't do anything (as described in https://github.com/cloudfoundry/bosh-agent/blob/8d18e4294e235edeb7f7fefc2f081ce006f57a71/vendor/github.com/charlievieth/fs/fs.go#L90-L97). That means the `userRootOptDirPermissions` won't be applied. Should we do an explicit `chmod`?

Reply:
Good catch, we apparently totally failed in our copy/paste from the /var/log code and didn't get the error handling. We just added that in a new commit.

We did look at the MkdirAll originally and opted not to do a `chmod` because we should be the only people making that path, so if it was made previously, it should have been made with the correct permissions.

We went back and took another look at it because we thought it might be possible for the `MkdirAll` to fail in the middle; make the director, but then the agent crashes before it can `chmod` it, and now the VM is in a bad state. But it seems like that's not possible, the mkdir syscall (way down in the call chain) takes both the directory name to create, and the permissions to give that directory, so I'm guessing it's an atomic operation within Linux.

Reply:
It sounds reasonable. @jpalermo  Thank you for the detailed consideration.